gpameter.user.js

```diff
@@ -27,9 +27,15 @@ var grade_values = {
 };
 console.log(studentId);
 add_checkboxes = function(){
+	var mapElems = new Map();
 	$(".cgpa_cal_check").remove();
-	elems = $(".hierarchyLi.dataLi").not(".hierarchyHdr, .hierarchySubHdr");	
+	elems = $(".hierarchyLi.dataLi").not(".hierarchyHdr, .hierarchySubHdr");
 	elems.each(function(i){
+		var cname = $(this).children(".col2").html().trim();
+		if (mapElems.has(cname)){
+			$(this).children(".col1").append("<input class=\"cgpa_cal_check\" type=\"checkbox\" />");
```

Comment:
Since we are doing the same thing below in line 49, how about moving this to a function, which adds a checkbox with a given value? Let's prefer reusing the code instead of copy-pasting it.
The function can take the element and a boolean (indicating checked or not) as input.

That also simplifies the code below to check boolean value instead of doing string comparison.